pypdf/filters.py

```diff
@@ -781,6 +781,9 @@ def _xobj_to_image(x_object_obj: Dict[str, Any]) -> Tuple[Optional[str], bytes,
     alpha = None
     filters = x_object_obj.get(SA.FILTER, [None])
     lfilters = filters[-1] if isinstance(filters, list) else filters
+    if isinstance(lfilters, IndirectObject):
+        lfilters = lfilters.get_object()
+        lfilters = lfilters[-1] if isinstance(lfilters, list) else lfilters
```

Comment:
```suggestion
    filters = x_object_obj.get(SA.FILTER, [None]).get_object()
    lfilters = filters[-1] if isinstance(filters, list) else filters
```
This should be clearer

Reply:
This will most likely break:

* Can we ensure that an existing `SA.FILTER` entry always is a *PdfObject*?
* `[None].get_object()` will always fail, id est the default case when we do not have an `SA.FILTER` entry.

Reply:
> This will most likely break:
> 
>     * Can we ensure that an existing `SA.FILTER` entry always is a _PdfObject_?
PdfObjects only returns PdfObjects si  SA.FILTER exists
> 
>     * `[None].get_object()` will always fail, id est the default case when we do not have an `SA.FILTER` entry.
Although Most probably exist in the current context, SA.FILTER may be ommited.
The best should be to return NullObject() instead, but also to consider this case in the filters.

Reply:
Thanks for the explanations. I have applied the proposed change, only changing the `[None]` to `NullObject()` to avoid having to add special handling for `get_object()`.